### ISSUE TITLE
Display locked domain behind login field?

### ISSUE THREAD

**Issue by poVoq** [conversejs/converse.js#1521]
When setting "locked_domain" is it somehow possible to display that in the login screen by adding the locked domain behind the input field?

Especially since many BOSH servers (such as the one build into Ejabberd) don't allow external JIDs anyways, the "locked_domain" (or "default_domain") seems like a very common/useful setting and it would be even better if the login page would clearly reflect this fact.

Thanks!

**Comment by jcbrand** [comment]
@nishitiwari22: There's no need to assign tickets, the chance that someone else will work on it in parallel to you is very slim, and in the past I've assigned tickets to people never to hear from them again.

If you'd like to work on this ticket, you can just start and then make a pull request once you're done.

**Comment by amar-jondhalekar** [comment]
Yes, it’s definitely possible to display the `locked_domain` on the login screen by modifying the input field for the username. Here’s how you can do it:

1. **Find the Login Template:** Look for the file in the codebase where the login form is defined.

2. **Update the Username Input:** Change the placeholder for the username input field to include the `locked_domain`. For example:

   ```html
   <input type="text" id="username" placeholder="Enter your username@locked_domain" />
   ```

3. **Fetch Dynamically:** Make sure to pull the `locked_domain` dynamically from the backend settings so it stays up to date.

4. **Test It:** Once you make these changes, test the login page to see if the `locked_domain` shows correctly and doesn’t mess with user input.

**Comment by poVoq** [comment]
Great, AI spam on github issues 💩

**Comment by Arup3201** [comment]
I am starting to work on this feature. While I understood what the feature request is, I still want to verify whether I understood it correctly or not...

So the feature request goes like this -

When initializing the converse.js using `converse.initialize` we can pass the `locked_domain` setting to it. After this, only users who has account on that domain can login.

Coming to the feature, you want to display the locked_domain name on the login screen when user inputs any domain user name.

Is this correct?

@jcbrand @poVoq

**Comment by poVoq** [comment]
Yes.

Other webclients such as Movim display the locked domain name is light grey in the input field so that people only need to type the name part of the JID which comes before the @.

**Comment by jcbrand** [comment]
Since we're using Bootstrap, here's an example:
https://getbootstrap.com/docs/5.3/forms/input-group/#basic-example

**Comment by Arup3201** [comment]
After working on it for some time, I got the following result:

![Image](https://github.com/user-attachments/assets/3fe8ea89-922c-4b2c-9da5-53ce6fd9c42e)

Is this UI good enough?

**Comment by jcbrand** [comment]
Yes, looks good to me.

**Comment by Arup3201** [comment]
I have raised a PR where you can find the tests with the code. I am finding issues when testing the login. It fails with the `waitUntill Promise timed out`. It comes from the following line: 
```js
await u.waitUntil(() => cbview.querySelectorAll('div.input-group').length);
```

I have tried the following things -
1. I checked the classname matches with what I have specified.
2. When printing the element, it gave me empty which shows that it has not found any element.
3. I have checked the authentication value is 'login' which should render the `tplAuthFields`.
4. I have put some console statements inside the template code to check whether we are going inside them or not, but nothing was printed.

PR: [ui: display locked domain behind login field](https://github.com/conversejs/converse.js/pull/3687)

Can you help me out on this?

**Comment by jcbrand** [comment]
Enable the debugger in the test and step through it line by line, starting at the first line. You'll likely find that `.input-group` element isn't in the DOM, and likely also why.

**Comment by jcbrand** [comment]
Hi @Arup3201, I had a quick look and there is no `input-group` class in the login form. Look for a different class you can query for. The point of that `waitUntil` line is to wait until the form has been rendered.

So you could try `await u.waitUntil(() => cbview.querySelector('.converse-login'));`

**Comment by Arup3201** [comment]
Yesterday I was busy so I could not progress on that. I will look into them today. I will also check the PR comments. Thanks for the help😄.

**Comment by Arup3201** [comment]
Hi @jcbrand I am trying to run the test using `make test` but getting an error in the `src/headless/plugins/caps/tests/caps.js` file. It says mock is not defined. Any idea on this? I am not able to run the test for login.

**Comment by Arup3201** [comment]
I tried the `make test` again, but it seems to run fine. I think I did something wrong while pulling and making changes from upstream branch.

**Comment by Arup3201** [comment]
Hi @jcbrand I have raised a new PR from another branch after pulling all new changes and updating the things you mentioned. Here is the PR: [ui: display locked_domain behind login field [tested]](https://github.com/conversejs/converse.js/pull/3691)

**Comment by Arup3201** [comment]
Hi @jcbrand @poVoq I am really glad that the PR merged, but a little bit unhappy that there is one mistake in it😅. I would like to update it and raise a PR, what do you think?

Also what are the other types of issues I could look into after this? There are a lot and I am a little lost there. Maybe you can help me in this?

Thanks. Really appreciate your active responses and efforts!

**Comment by jcbrand** [comment]
Hi @Arup3201, here a few you can look at:

- https://github.com/conversejs/converse.js/issues/1855
- https://github.com/conversejs/converse.js/issues/2692
- https://github.com/conversejs/converse.js/issues/3120
- https://github.com/conversejs/converse.js/issues/3023

You can also look at the `Good first issue` tag: https://github.com/conversejs/converse.js/issues?q=state%3Aopen%20label%3A%22good%20first%20issue%22

If you'd like to tackle one of these issues, let me know and I'll try to provide some more context and info.
